<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME)" />

  <group ns="$(arg UAV_NAME)">

    <!-- SENSOR TOPICS -->
    <arg name="imu_topic" default="vio/imu_filtered" />
    <arg name="image_topic" default="/mv_25003659/image_raw" />

    <arg name="config_path" default = "$(find feature_tracker)/../config/bluefox/bluefox_icm_4688.yaml" />
    <arg name="vins_path" default = "$(find feature_tracker)/../config/../" />
    
    <node name="feature_tracker" pkg="feature_tracker" type="feature_tracker" output="log">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="vins_folder" type="string" value="$(arg vins_path)" />

        <!-- subscribers -->
        <remap from="~image_in" to="$(arg image_topic)" />

    </node>

    <node name="vins_estimator" pkg="vins_estimator" type="vins_estimator" output="screen">
       <param name="config_file" type="string" value="$(arg config_path)" />
       <param name="vins_folder" type="string" value="$(arg vins_path)" />

        <!-- subscribers -->
        <remap from="~imu_in" to="$(arg imu_topic)" />
        <remap from="~feature_tracker/feature" to="feature_tracker/feature" />
        <remap from="~feature_tracker/restart" to="feature_tracker/restart" />
        <remap from="~pose_graph/match_points" to="pose_graph/match_points" />

    </node>

    <node name="pose_graph" pkg="pose_graph" type="pose_graph" output="screen">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="visualization_shift_x" type="int" value="0" />
        <param name="visualization_shift_y" type="int" value="0" />
        <param name="skip_cnt" type="int" value="0" />
        <param name="skip_dis" type="double" value="0" />

        <!-- subscribers -->
        <remap from="~image_in" to="$(arg image_topic)" />
        <remap from="~vins_estimator/imu_propagate" to="vins_estimator/imu_propagate" />
        <remap from="~vins_estimator/odometry" to="vins_estimator/odometry" />
        <remap from="~vins_estimator/keyframe_pose" to="vins_estimator/keyframe_pose" />
        <remap from="~vins_estimator/keyframe_point" to="vins_estimator/keyframe_point" />
        <remap from="~vins_estimator/relo_relative_pose" to="vins_estimator/relo_relative_pose" />
        <remap from="~vins_estimator/extrinsic" to="vins_estimator/extrinsic" />

    </node>

  </group>

</launch>
